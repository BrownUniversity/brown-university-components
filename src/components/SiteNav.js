import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styled, { css } from 'styled-components';
import { WindowSize } from 'react-fns';
import { Collapse } from 'react-collapse';

import Hamburger from './Hamburger';
import Nav from './Nav';
import breakpoints from '../constants/breakpoints';
import colors from '../constants/colors';
import { sansBold } from '../constants/typography';

/*
  inner components
*/
const MobileBannerPositioningWrapper = styled.div`
  ${props =>
    props.banner &&
    css`
      background-color: ${colors.white};
      margin: 0 auto;
      margin-top: 2rem;
      position: relative;
      width: 95%;
      z-index: 10;
    `};
`;

const MobileWrapper = styled.nav`
  border: 1px solid #ddd;
`;

const MobileToggleButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  font-size: 1em;
  padding: 1rem 0 1rem;
  width: 100%;
`;

const MobileToggleButtonInner = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0 auto;
  width: 92%;
`;

const MobileToggleTitle = styled.span`
  color: ${colors.red};
  font-family: ${sansBold};
  font-weight: bold;
  letter-spacing: 0.5px;
  padding: 11px 5px;
  text-transform: uppercase;
`;

const MobileNavWrapper = styled.nav`
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 0 auto;
  padding: 16px 0;
  width: 92%;
  z-index: 10;
`;

const BannerPositioningWrapper = styled.div`
  background-color: ${colors.white};
  ${props =>
    props.banner &&
    css`
      margin: 0 auto;
      margin-top: -50px;
      max-width: 1300px;
      position: relative;
      width: 96%;
      z-index: 10;
    `}
`;

const NavWrapper = styled.div`
  display: flex;
  justify-content: center;
  padding: 20px 0 30px;
`;

/*
  outer SiteNav component
*/
class SiteNav extends Component {
  state = {
    mobileNavIsOpen: false
  };

  handleMobileNavToggle = () =>
    this.setState(({ mobileNavIsOpen }) => ({
      mobileNavIsOpen: !mobileNavIsOpen
    }));

  render() {
    const {
      banner,
      mobileBreakpoint,
      mobileToggleTitle,
      ...restProps
    } = this.props;
    const { mobileNavIsOpen } = this.state;

    return (
      <WindowSize
        render={({ width }) => {
          // TODO: update when `width` doesn't return 0 on initial render (see `react-fns` issue 84)
          const currentWidth = width === 0 ? window.innerWidth : width;
          const renderMobile = currentWidth < mobileBreakpoint;

          if (renderMobile) {
            return (
              <MobileBannerPositioningWrapper banner={banner}>
                <MobileWrapper {...restProps}>
                  <MobileToggleButton
                    type="button"
                    aria-controls="site-nav-mobile-collapse"
                    aria-expanded={mobileNavIsOpen}
                    aria-label="Toggle site navigation"
                    onClick={this.handleMobileNavToggle}
                  >
                    <MobileToggleButtonInner>
                      <MobileToggleTitle>{mobileToggleTitle}</MobileToggleTitle>
                      <Hamburger tag="div" isOpen={mobileNavIsOpen} />
                    </MobileToggleButtonInner>
                  </MobileToggleButton>
                  <Collapse
                    id="site-nav-mobile-collapse"
                    isOpened={mobileNavIsOpen}
                  >
                    <MobileNavWrapper>
                      <Nav mobile>{this.props.children}</Nav>
                    </MobileNavWrapper>
                  </Collapse>
                </MobileWrapper>
              </MobileBannerPositioningWrapper>
            );
          }

          return (
            <BannerPositioningWrapper banner={banner}>
              <NavWrapper {...restProps}>
                <Nav>{this.props.children}</Nav>
              </NavWrapper>
            </BannerPositioningWrapper>
          );
        }}
      />
    );
  }
}

SiteNav.propTypes = {
  banner: PropTypes.bool,
  mobileBreakpoint: PropTypes.number,
  mobileToggleTitle: PropTypes.string,
  children: PropTypes.node.isRequired
};

SiteNav.defaultProps = {
  banner: false,
  mobileBreakpoint: breakpoints.md,
  mobileToggleTitle: 'Site Navigation'
};

SiteNav.Item = Nav.Item;
SiteNav.Link = Nav.Link;

export default SiteNav;
