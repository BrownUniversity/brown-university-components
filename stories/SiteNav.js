import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, number, text } from '@storybook/addon-knobs';

import { SiteNav, Banner } from '../src';
import breakpoints from '../src/constants/breakpoints';
import bannerImage from './images/banner.png';

const getCommonProps = () => ({
  mobileMenuTitle: text('mobileMenuTitle', 'Site Navigation'),
  mobileNavBreakpoint: number('mobileNavBreakpoint', breakpoints.md)
});

const renderSiteNav = additionalProps => (
  <SiteNav {...getCommonProps()} {...additionalProps}>
    <SiteNav.Item>
      <SiteNav.Link>Link</SiteNav.Link>
    </SiteNav.Item>
    <SiteNav.Item>
      <SiteNav.Link active>Active Link</SiteNav.Link>
    </SiteNav.Item>
    <SiteNav.Item>
      <SiteNav.Link disabled>Disabled Link</SiteNav.Link>
    </SiteNav.Item>
  </SiteNav>
);
storiesOf('SiteNav', module)
  .addDecorator(withKnobs)
  .add('default', () => renderSiteNav())
  .add('with banner', () => (
    <React.Fragment>
      <Banner src={bannerImage}>
        <Banner.Text>University Theme</Banner.Text>
      </Banner>
      {renderSiteNav({ banner: boolean('banner', true) })}
    </React.Fragment>
  ));
