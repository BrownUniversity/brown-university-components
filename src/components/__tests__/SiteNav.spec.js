import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { triggerWindowResize } from 'window-test-utils';

import SiteNav from '../SiteNav';
import breakpoints from '../../constants/breakpoints';

const renderSiteNav = ({
  props = {},
  children = (
    <React.Fragment>
      <SiteNav.Item>
        <SiteNav.Link>Link</SiteNav.Link>
      </SiteNav.Item>
      <SiteNav.Item>
        <SiteNav.Link active>Active Link</SiteNav.Link>
      </SiteNav.Item>
      <SiteNav.Item>
        <SiteNav.Link disabled>Link</SiteNav.Link>
      </SiteNav.Item>
    </React.Fragment>
  )
} = {}) => {
  const rtlUtils = render(<SiteNav {...props}>{children}</SiteNav>);

  return {
    siteNav: rtlUtils.container.firstChild,
    ...rtlUtils
  };
};

describe('SiteNav', () => {
  describe('when window width is above mobileBreakpoint', () => {
    describe('style', () => {
      describe('when banner variant is not provided', () => {
        it('should render default site nav', () => {
          const { siteNav } = renderSiteNav();

          expect(siteNav).toMatchSnapshot();
        });
      });

      describe('when banner variant is provided', () => {
        it('should render default site nav with banner positioning', () => {
          const { siteNav } = renderSiteNav({ props: { banner: true } });

          expect(siteNav).toMatchSnapshot();
        });
      });
    });
  });

  describe('when window width is below mobileBreakpoint', () => {
    beforeEach(() => {
      triggerWindowResize({ width: breakpoints.md - 1 });
    });

    describe('style', () => {
      describe('when banner variant is not provided', () => {
        it('should render mobile site nav', () => {
          const { siteNav } = renderSiteNav();

          expect(siteNav).toMatchSnapshot();
        });
      });

      describe('when banner variant is provided', () => {
        it('should render mobile site nav with banner positioning', () => {
          const { siteNav } = renderSiteNav({ props: { banner: true } });

          expect(siteNav).toMatchSnapshot();
        });
      });
    });

    it('should render custom mobile toggle title if provided', () => {
      const { getByText } = renderSiteNav({
        props: { mobileToggleTitle: 'Custom Navigation' }
      });

      expect(getByText('Custom Navigation')).toBeInTheDocument();
    });

    it('should toggle site nav mobile collapse', () => {
      const { getByLabelText, getByText } = renderSiteNav();
      const mobileToggleButtonByLabelText = getByLabelText(
        'Toggle site navigation'
      );
      const mobileToggleButtonByText = getByText('Site Navigation');

      expect(mobileToggleButtonByLabelText).toHaveAttribute(
        'aria-expanded',
        'false'
      );
      fireEvent.click(mobileToggleButtonByLabelText);
      expect(mobileToggleButtonByLabelText).toHaveAttribute(
        'aria-expanded',
        'true'
      );
      fireEvent.click(mobileToggleButtonByText);
      expect(mobileToggleButtonByLabelText).toHaveAttribute(
        'aria-expanded',
        'false'
      );
    });
  });
});
