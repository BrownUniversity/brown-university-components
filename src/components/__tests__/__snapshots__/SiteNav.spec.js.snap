// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteNav when window width is above mobileBreakpoint styles when banner variant is not provided should render default site nav 1`] = `
.c3 {
  display: list-item;
  padding: 9px 15px;
}

.c4 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #3B302C;
  cursor: pointer;
  font-size: 1.1em;
  font-weight: 700;
  line-height: 1;
  opacity: 1;
  pointer-events: auto;
}

.c4::after {
  content: '';
  display: block;
  height: 3px;
  background: #C00404;
  margin-top: 1em;
  width: 0;
}

.c4:hover {
  color: #C00404;
}

.c4:hover::after {
  width: 100%;
}

.c5 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: pointer;
  font-size: 1.1em;
  font-weight: 700;
  line-height: 1;
  opacity: 1;
  pointer-events: auto;
}

.c5::after {
  content: '';
  display: block;
  height: 3px;
  background: #C00404;
  margin-top: 1em;
  width: 100%;
}

.c5:hover {
  color: #C00404;
}

.c5:hover::after {
  width: 100%;
}

.c6 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #3B302C;
  cursor: not-allowed;
  font-size: 1.1em;
  font-weight: 700;
  line-height: 1;
  opacity: 0.65;
  pointer-events: auto;
}

.c6::after {
  content: '';
  display: block;
  height: 3px;
  background: #C00404;
  margin-top: 1em;
  width: 0;
}

.c6:hover {
  color: #3B302C;
}

.c6:hover::after {
  width: 0;
}

.c2 {
  list-style: none;
  margin: 0;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  width: auto;
}

.c0 {
  background-color: #FFFFFF;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0 30px;
}

@media (min-width:1200px) {
  .c4 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c5 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c6 {
    font-size: 1.2em;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <button
          class="c4"
          color="red"
        >
          Link
        </button>
      </li>
      <li
        class="c3"
      >
        <button
          class="c5"
          color="red"
        >
          Active Link
        </button>
      </li>
      <li
        class="c3"
      >
        <button
          class="c6"
          color="red"
          disabled=""
        >
          Link
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SiteNav when window width is above mobileBreakpoint styles when banner variant is provided should render default site nav with banner positioning 1`] = `
.c3 {
  display: list-item;
  padding: 9px 15px;
}

.c4 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #3B302C;
  cursor: pointer;
  font-size: 1.1em;
  font-weight: 700;
  line-height: 1;
  opacity: 1;
  pointer-events: auto;
}

.c4::after {
  content: '';
  display: block;
  height: 3px;
  background: #C00404;
  margin-top: 1em;
  width: 0;
}

.c4:hover {
  color: #C00404;
}

.c4:hover::after {
  width: 100%;
}

.c5 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: pointer;
  font-size: 1.1em;
  font-weight: 700;
  line-height: 1;
  opacity: 1;
  pointer-events: auto;
}

.c5::after {
  content: '';
  display: block;
  height: 3px;
  background: #C00404;
  margin-top: 1em;
  width: 100%;
}

.c5:hover {
  color: #C00404;
}

.c5:hover::after {
  width: 100%;
}

.c6 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #3B302C;
  cursor: not-allowed;
  font-size: 1.1em;
  font-weight: 700;
  line-height: 1;
  opacity: 0.65;
  pointer-events: auto;
}

.c6::after {
  content: '';
  display: block;
  height: 3px;
  background: #C00404;
  margin-top: 1em;
  width: 0;
}

.c6:hover {
  color: #3B302C;
}

.c6:hover::after {
  width: 0;
}

.c2 {
  list-style: none;
  margin: 0;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  width: auto;
}

.c0 {
  background-color: #FFFFFF;
  margin: 0 auto;
  margin-top: -50px;
  max-width: 1300px;
  position: relative;
  width: 96%;
  z-index: 10;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0 30px;
}

@media (min-width:1200px) {
  .c4 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c5 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c6 {
    font-size: 1.2em;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <button
          class="c4"
          color="red"
        >
          Link
        </button>
      </li>
      <li
        class="c3"
      >
        <button
          class="c5"
          color="red"
        >
          Active Link
        </button>
      </li>
      <li
        class="c3"
      >
        <button
          class="c6"
          color="red"
          disabled=""
        >
          Link
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SiteNav when window width is below mobileBreakpoint styles when banner variant is not provided should render mobile site nav 1`] = `
.c4 {
  background: transparent;
  border: none;
  cursor: inherit;
  display: inline-block;
  padding: 0 25px 7px 0;
}

.c5 {
  height: 3px;
  position: absolute;
  width: 25px;
  background-color: #C00404;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  -webkit-transition-duration: .22s;
  transition-duration: .22s;
  -webkit-transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
  transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
}

.c5.c5:before {
  height: 3px;
  position: absolute;
  width: 25px;
  background-color: #C00404;
  content: '';
  display: block;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  -webkit-transition-duration: .15s;
  transition-duration: .15s;
  -webkit-transition: top .1s .25s ease-in,opacity .1s ease-in;
  transition: top .1s .25s ease-in,opacity .1s ease-in;
  top: -7px;
}

.c5.c5:after {
  height: 3px;
  position: absolute;
  width: 25px;
  background-color: #C00404;
  content: '';
  display: block;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  -webkit-transition-duration: .15s;
  transition-duration: .15s;
  -webkit-transition: bottom .1s .25s ease-in,-webkit-transform .22s cubic-bezier(0.55,0.055,0.675,0.19);
  -webkit-transition: bottom .1s .25s ease-in,transform .22s cubic-bezier(0.55,0.055,0.675,0.19);
  transition: bottom .1s .25s ease-in,transform .22s cubic-bezier(0.55,0.055,0.675,0.19);
  bottom: -7px;
}

.c8 {
  display: list-item;
  border-top: 0.5px solid #C8C8C880;
  padding: 1rem 0;
}

.c9 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: pointer;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;
  opacity: 1;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c9:hover {
  color: #3B302C;
}

.c10 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #3B302C;
  cursor: pointer;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;
  opacity: 1;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c10:hover {
  color: #3B302C;
}

.c11 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: not-allowed;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;
  opacity: 0.65;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c11:hover {
  color: #C00404;
}

.c7 {
  list-style: none;
  margin: 0;
  background-color: transparent;
  display: block;
  padding: 0;
  width: 100%;
}

.c0 {
  border: 1px solid #ddd;
}

.c1 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  font-size: 1em;
  padding: 1rem 0 1rem;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto;
  width: 92%;
}

.c3 {
  color: #C00404;
  font-family: Circular-Bold,Circular-Book,Arial,Helvetica,sans-serif;
  font-weight: bold;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 11px 5px;
  text-transform: uppercase;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  padding: 16px 0;
  width: 92%;
  z-index: 10;
}

<div
  class=""
>
  <nav
    class="c0"
  >
    <button
      aria-controls="site-nav-mobile-collapse"
      aria-expanded="false"
      aria-label="Toggle site navigation"
      class="c1"
      type="button"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          Site Navigation
        </span>
        <div
          class="c4"
        >
          <span
            class="c5"
          />
        </div>
      </div>
    </button>
    <div
      class="ReactCollapse--collapse"
      id="site-nav-mobile-collapse"
      style="overflow: hidden; height: 0px; display: none;"
    >
      <div
        class="ReactCollapse--content"
      >
        <div
          class="c6"
        >
          <ul
            class="c7"
          >
            <li
              class="c8"
            >
              <button
                class="c9"
                color="red"
              >
                Link
              </button>
            </li>
            <li
              class="c8"
            >
              <button
                class="c10"
                color="red"
              >
                Active Link
              </button>
            </li>
            <li
              class="c8"
            >
              <button
                class="c11"
                color="red"
                disabled=""
              >
                Link
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`SiteNav when window width is below mobileBreakpoint styles when banner variant is provided should render mobile site nav with banner positioning 1`] = `
.c5 {
  background: transparent;
  border: none;
  cursor: inherit;
  display: inline-block;
  padding: 0 25px 7px 0;
}

.c6 {
  height: 3px;
  position: absolute;
  width: 25px;
  background-color: #C00404;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  -webkit-transition-duration: .22s;
  transition-duration: .22s;
  -webkit-transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
  transition-timing-function: cubic-bezier(0.55,0.055,0.675,0.19);
}

.c6.c6:before {
  height: 3px;
  position: absolute;
  width: 25px;
  background-color: #C00404;
  content: '';
  display: block;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  -webkit-transition-duration: .15s;
  transition-duration: .15s;
  -webkit-transition: top .1s .25s ease-in,opacity .1s ease-in;
  transition: top .1s .25s ease-in,opacity .1s ease-in;
  top: -7px;
}

.c6.c6:after {
  height: 3px;
  position: absolute;
  width: 25px;
  background-color: #C00404;
  content: '';
  display: block;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
  -webkit-transition-duration: .15s;
  transition-duration: .15s;
  -webkit-transition: bottom .1s .25s ease-in,-webkit-transform .22s cubic-bezier(0.55,0.055,0.675,0.19);
  -webkit-transition: bottom .1s .25s ease-in,transform .22s cubic-bezier(0.55,0.055,0.675,0.19);
  transition: bottom .1s .25s ease-in,transform .22s cubic-bezier(0.55,0.055,0.675,0.19);
  bottom: -7px;
}

.c9 {
  display: list-item;
  border-top: 0.5px solid #C8C8C880;
  padding: 1rem 0;
}

.c10 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: pointer;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;
  opacity: 1;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c10:hover {
  color: #3B302C;
}

.c11 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #3B302C;
  cursor: pointer;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;
  opacity: 1;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c11:hover {
  color: #3B302C;
}

.c12 {
  background: transparent;
  border: none;
  display: block;
  font-family: Circular-Book,Arial,Helvetica,sans-serif;
  padding: 0;
  -webkit-text-decoration: none !important;
  text-decoration: none !important;
  color: #C00404;
  cursor: not-allowed;
  font-size: 1em;
  font-weight: 700;
  line-height: 1.5;
  opacity: 0.65;
  pointer-events: auto;
  -webkit-transition: all .2s ease-in-out;
  transition: all .2s ease-in-out;
}

.c12:hover {
  color: #C00404;
}

.c8 {
  list-style: none;
  margin: 0;
  background-color: transparent;
  display: block;
  padding: 0;
  width: 100%;
}

.c0 {
  background-color: #FFFFFF;
  margin: 0 auto;
  margin-top: 2rem;
  position: relative;
  width: 95%;
  z-index: 10;
}

.c1 {
  border: 1px solid #ddd;
}

.c2 {
  background-color: transparent;
  border: none;
  cursor: pointer;
  font-size: 1em;
  padding: 1rem 0 1rem;
  width: 100%;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 auto;
  width: 92%;
}

.c4 {
  color: #C00404;
  font-family: Circular-Bold,Circular-Book,Arial,Helvetica,sans-serif;
  font-weight: bold;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 11px 5px;
  text-transform: uppercase;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  padding: 16px 0;
  width: 92%;
  z-index: 10;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <button
      aria-controls="site-nav-mobile-collapse"
      aria-expanded="false"
      aria-label="Toggle site navigation"
      class="c2"
      type="button"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          Site Navigation
        </span>
        <div
          class="c5"
        >
          <span
            class="c6"
          />
        </div>
      </div>
    </button>
    <div
      class="ReactCollapse--collapse"
      id="site-nav-mobile-collapse"
      style="overflow: hidden; height: 0px; display: none;"
    >
      <div
        class="ReactCollapse--content"
      >
        <div
          class="c7"
        >
          <ul
            class="c8"
          >
            <li
              class="c9"
            >
              <button
                class="c10"
                color="red"
              >
                Link
              </button>
            </li>
            <li
              class="c9"
            >
              <button
                class="c11"
                color="red"
              >
                Active Link
              </button>
            </li>
            <li
              class="c9"
            >
              <button
                class="c12"
                color="red"
                disabled=""
              >
                Link
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
